src/main/java/org/hydev/mcpm/client/injector/LocalJarFinder.java

```diff
@@ -0,0 +1,40 @@
+package org.hydev.mcpm.client.injector;
+
+import org.hydev.mcpm.client.models.PluginYml;
+import org.hydev.mcpm.utils.PluginJarFile;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Optional;
+
+/**
+ * Implementation that finds locally installed jars by name
+ *
+ * @param dir Plugins directory (default to "./plugins")
+ * @author Azalea (https://github.com/hykilpikonna)
+ * @since 2022-11-27
+ */
+public record LocalJarFinder(File dir) implements LocalJarBoundary
```

Comment:
Did this make it into here on purpose? Seems like the history for this PR is weird. Not sure if you pulled these classes in because you needed them or something.